db/migrate/20200326231610_add_slug_to_reading_collections.rb

```diff
@@ -0,0 +1,5 @@
+class AddSlugToReadingCollections < ActiveRecord::Migration[5.2]
+  def change
+    add_column :reading_collections, :slug, :string
```

Comment:
So to gain understanding here, this and the other slug change below just implement specific handling to generate the slug dynamically?

Reply:
Exactly. When a reading collection is created, it needs to have a "slug" also created. It's like an alias so you don't have to display a resources id in a route (or anywhere). It's a unique string based on the name of the reading collection. The "before_validation" creates the slug so the reading collection can be created without a human entering a slug.